### PR TITLE
Add zenodo crawler script (in progress)

### PR DESCRIPTION
**Pull request by joeyzhou98**
## Description
Instead of going through the complex procedure of uploading a dataset themselves, the dataset owners can now upload their dataset onto Zenodo with the keyword `canadian-open-neuroscience-platform` and this script will automatically fetch new changes and versions and create a pull request to this repository.

The procedure is as follows:
1. Make sure you have an up to date repository by pulling
2. Retrieve all current conp-dataset information stored in dats.json files (concept_doi and version_doi)
3. Retrieve all metadata for all zenodo datasets containing the keyword `canadian-open-neuroscience-platform`
4. Looping through the retrieved zenodo datasets metadata:
    1. If zenodo dataset is existing in a conp-dataset by comparing concept_doi:
        1. If the conp-dataset version is not current by comparing the zenodo dataset latest_version_doi with the conp-dataset version_doi:
            1. Update conp-dataset
    2. If the zenodo dataset doesn't exist in conp-dataset by seeing that the zenodo concept_doi is not in the concept_dois retrieved from conp-datasets
        1. Download zenodo dataset into a new dataset/directory
5. Create pull requests with changes

Here is some documentation regarding [concept_dois](http://help.zenodo.org/#versioning)

Issue #79

### CODE REVIEW

**Comment by glatard** [comment]
Hey @joeyzhou98, nice work! One thing that should absolutely be added are tests, as this is going to be a sensitive piece of code. You should add it to the Travis file you created before.

File: scripts/crawl_zenodo.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        zenodo_dois.append({
+            "concept_doi": concept_doi,
+            "latest_version": latest_version_doi
+        })
+
+    return zenodo_dois
+
+
+def verify_duplicates(zenodo_dois, conp_dois):
+    concept_dois = list(map(lambda x: x["concept_doi"], zenodo_dois))
```
Comment (jbpoline) [inline scripts/crawl_zenodo.py]: some docstring here ?

File: scripts/crawl_zenodo.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    if len(concept_dois) != len(set(concept_dois)):
+        raise Exception("Concept DOI duplicates exists in zenodo list")
+    version_dois = list(map(lambda x: x["latest_version"], zenodo_dois))
+    if len(version_dois) != len(set(version_dois)):
+        raise Exception("Version DOI duplicates exists in zenodo list")
+    concept_dois = list(map(lambda x: x["concept_doi"], conp_dois))
+    if len(concept_dois) != len(set(concept_dois)):
+        raise Exception("Concept DOI duplicates exists in conp list")
+    version_dois = list(map(lambda x: x["version"], conp_dois))
```
Comment (jbpoline) [inline scripts/crawl_zenodo.py]: version_dois is also = list(map(lambda x: x["latest_version"], zenodo_dois)) ?

Comment (joeyzhou98) [reply scripts/crawl_zenodo.py]: I could rename it to another variable but in the end it won't affect the verification

**Comment by joeyzhou98** [comment]
@PapillonMcGill the PR wasn't ready yet, but it shouldn't impact anyone's code. I will let you know when it is ready for review

**Comment by PapillonMcGill** [comment]
@joeyzhou98 I can revert the merge if needed.

**Pull request merged**